app/src/test/java/com/nvapp/navigation/route/RouteLinkFactoryTest.java

```diff
@@ -0,0 +1,62 @@
+package com.nvapp.navigation.route;
+
+import android.net.Uri;
+
+import com.google.maps.model.LatLng;
+import com.navapp.navigation.destination.Destination;
+import com.navapp.navigation.destination.data.Location;
+import com.navapp.navigation.route.Route;
+import com.navapp.navigation.route.RouteLinkFactory;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+import java.util.ArrayList;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(manifest = Config.NONE)
+public class RouteLinkFactoryTest {
+    private static final String NAVIGATION_LINK =  "https://www.google.com/maps/dir/?api=1";
+    private static final String ORIGIN_TEMPLATE = "&origin=%s&origin_place_id=%s";
+    private static final String DESTINATION_TEMPLATE = "&destination=%s&destination_place_id=%s";
+    private static final String WAYPOINT_TEMPLATE = "&waypoints=%s|%s&waypoint_place_ids=%s|%s";
+
+    private static final String DRIVING_TRAVEL_MODE = "&travelmode=driving";
+    private static final String SKIP_OVERVIEW = "&dir_action=navigate";
+
+    private Route route;
+    private String baseExpectedLinkText;
+
+    @Before
+    public void createRoute() {
```

Comment:
Do https://www.testwithspring.com/lesson/writing-parameterized-tests-with-junit-4/

Reply:
can you explain how should I parameterize it? and why?

Reply:
I mean, why isn't this good enough?

Reply:
by explain, which part should I parameterize? (besides everything)

Reply:
will add this to route link builder test